package/MDAnalysis/analysis/base.py

```diff
@@ -125,14 +125,105 @@
+                raise ValueError(f"client should be dask.distributed.Client instance, got {type(client)}")
+        self.client = client
+
+        # validate 'backend' argument
+        if backend is not None:
+            if backend not in ParallelExecutor.possible_backends:
+                raise ValueError(
+                    f"Invalid {backend=}: should be one of {ParallelExecutor.possible_backends}"
+                )
+            if backend == "local":
+                if n_workers > 1:
+                    raise ValueError("Can be only 1 worker with backend='local', got {n_workers=")
+            else:
+                try:
+                    import_module(backend)
+                except ImportError:
+                    raise ValueError(
+                        "Please install {backend} module "
+                        "with 'python3 -m pip install {backend} "
+                        "to run this")
+        self.backend = backend
+
+        # validate 'n_workers' argument
+        if n_workers and n_workers < 0:
+            raise ValueError(f"n_workers should be non-negative, got {n_workers=}")
+        self.n_workers = n_workers
+
+    def apply(self, func: Callable, computations: list) -> list:
+        if self.backend == "local":
+            return self._compute_with_local(func, computations)
+        elif self.backend == "multiprocessing":
+            return self._compute_with_multiprocessing(func, computations)
+        elif self.backend == "dask":
+            return self._compute_with_dask(func, computations)
+        elif self.client:
+            return self._compute_with_client(func, computations)
+        elif self.backend == 'dask.distributed':
+            raise ValueError(f"To use this backend, pre-configure a Client object and pass client=... instead")
+        else:
+            raise ValueError(f"Backend or client is not set properly: {self.backend=}, {self.client=}")
+
+    def _compute_with_local(self, func: Callable, computations: list) -> list:
+        return [func(task) for task in computations]
+
+    def _compute_with_multiprocessing(self, func: Callable, computations: list) -> list:
+        from multiprocessing import Pool
+
+        with Pool(processes=self.n_workers) as pool:
+            results = pool.map(func, computations)
+        return results
+
+    def _compute_with_dask(self, func: Callable, computations: list) -> list:
+        from dask.delayed import delayed
+        import dask
+        from multiprocessing import Pool
+
+        computations = [delayed(func)(task) for task in computations]
+        with Pool(self.n_workers):
+            with dask.config.set(pool=Pool):
+                results = dask.compute(computations)[0]
+        return results
+    
+    def _compute_with_client(self, func: Callable, computations: list) -> list:
+        from dask.delayed import delayed
+        computations = delayed([delayed(func)(task) for task in computations])
```

Comment:
I see you removed the external `delayed` from `_compute_with_dask` but it is still present here. I'm not totally sure which one is best, but the common patter in Dask's documentation seems to avoid the external delayed and do `dask.compute(*computations)`.

```python
results = []
for x in L:
    y = dask.delayed(f)(x)
    results.append(y)

results = dask.compute(*results)
```

Reply:
Thanks, fixed.